components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2146,39 +2146,87 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    async getUsageLimitForTeam(ctx: TraceContext, teamId: string): Promise<number | undefined> {
-        const user = this.checkAndBlockUser("getUsageLimitForTeam");
-        const team = await this.guardTeamOperation(teamId, "get");
-        await this.ensureStripeApiIsAllowed({ team });
+    async getUsageLimit(ctx: TraceContext, attributionId: string): Promise<number | undefined> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId === undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
 
-        const attributionId: AttributionId = { kind: "team", teamId };
-        await this.guardCostCenterAccess(ctx, user.id, attributionId, "get");
+        const user = this.checkAndBlockUser("getUsageLimit");
+        switch (attrId.kind) {
+            case "team":
+                const team = await this.guardTeamOperation(attrId.teamId, "get");
+                await this.ensureStripeApiIsAllowed({ team });
+                break;
+            case "user":
+                await this.ensureStripeApiIsAllowed({ user });
+                break;
+        }
+        await this.guardCostCenterAccess(ctx, user.id, attrId, "get");
+
+        const costCenter = await this.usageServiceClientProvider.getDefault().getCostCenter(attrId);
+        if (costCenter?.billingStrategy !== "stripe") {
```

Comment:
@andrew-farries, if I got @svenefftinge correct, this would be still valid. We are going to map other strategies and assign usage limits internally. 

The guard in `setUsageLimit` looks good though!

Reply:
Yes, reading them is fine. Just not changing.